src/org/zaproxy/zap/extension/stats/OptionsStatsPanel.java

```diff
@@ -0,0 +1,215 @@
+	}
+
+	private JCheckBox getStatsdStatsEnabledField() {
+		if (statsdStatsEnabledField == null) {
+			statsdStatsEnabledField = new JCheckBox();
+			statsdStatsEnabledField.setText(Constant.messages.getString("stats.options.statsd.enabled"));
+			statsdStatsEnabledField.setVerticalAlignment(javax.swing.SwingConstants.TOP);
+			statsdStatsEnabledField.setVerticalTextPosition(javax.swing.SwingConstants.TOP);
+			statsdStatsEnabledField.addActionListener(new ActionListener() {
+
+				@Override
+				public void actionPerformed(ActionEvent e) {
+					setStatsdFieldStates();
+				}});
+
+		}
+		return statsdStatsEnabledField;
+	}
+
+	private void setStatsdFieldStates() {
+		getStatsdHostField().setEnabled(statsdStatsEnabledField.isSelected());
+		getStatsdPortField().setEnabled(statsdStatsEnabledField.isSelected());
+		getStatsdPrefixField().setEnabled(statsdStatsEnabledField.isSelected());
+	}
+
+	private ZapTextField getStatsdHostField() {
+		if (statsdHostField == null) {
+			statsdHostField = new ZapTextField();
+		}
+		return statsdHostField;
+	}
+
+	private ZapPortNumberSpinner getStatsdPortField() {
+		if (statsdPortField == null) {
+			statsdPortField = new ZapPortNumberSpinner(0);
+		}
+		return statsdPortField;
+	}
+
+	private ZapTextField getStatsdPrefixField() {
+		if (statsdPrefixField == null) {
+			statsdPrefixField = new ZapTextField();
+		}
+		return statsdPrefixField;
+	}
+
+	@Override
+    public void initParam(Object obj) {
+	    OptionsParam optionsParam = (OptionsParam) obj;
+		StatsParam statsParam = optionsParam.getParamSet(StatsParam.class);
+
+	    getInMemoryStatsEnabledField().setSelected(statsParam.isInMemoryEnabled());
+	    getStatsdStatsEnabledField().setSelected(statsParam.isStatsdEnabled());
+	    getStatsdHostField().setText(statsParam.getStatsdHost());
+	    getStatsdPortField().setValue(statsParam.getStatsdPort());
+	    getStatsdPrefixField().setText(statsParam.getStatsdPrefix());
+
+	    setStatsdFieldStates();
+    }
+
+
+    @Override
+    public void validateParam(Object obj) throws Exception {
+	    if (getStatsdStatsEnabledField().isSelected()) {
+	    	// Basic hostname validation
+	    	try {
+				InetAddress.getByName(getStatsdHostField().getText()); 
+			} catch (Exception e) {
+				throw new Exception(Constant.messages.getString("stats.options.error.statsd.host.bad"), e);
+			}
+	    }
+    }
+
+
+    @Override
+    public void saveParam(Object obj) throws Exception {
+	    OptionsParam optionsParam = (OptionsParam) obj;
+		StatsParam statsParam = optionsParam.getParamSet(StatsParam.class);
+
+	    statsParam.setInMemoryEnabled(getInMemoryStatsEnabledField().isSelected());
+	    
+	    if (getStatsdStatsEnabledField().isSelected()) {
+		    statsParam.setStatsdHost(getStatsdHostField().getText());
+		    statsParam.setStatsdPort(getStatsdPortField().getValue());
+		    statsParam.setStatsdPrefix(getStatsdPrefixField().getText());
+	    } else {
+		    statsParam.setStatsdHost(null);

```

Comment:
Setting to `null` leads to an internal error when attempting to see the host through the ZAP API (<http://zap/JSON/stats/view/optionStatsdHost/>).
Also, this requires setting the host again even if it's to just temporarily disable it (also changes done to port/prefix are not saved when done before deselecting the checkbox).